src/policy/space.rs

```diff
@@ -574,15 +574,6 @@ impl<VM: VMBinding> CommonSpace<VM> {
     pub fn init(&self, space: &dyn Space<VM>) {
         // For contiguous space, we eagerly initialize SFT map based on its address range.
         if self.contiguous {
-            if !try_map_metadata_space(
-                self.start,
-                self.extent,
-                VM::VMActivePlan::global().global_side_metadata_per_chunk(),
-                space.local_side_metadata_per_chunk(),
-            ) {
-                // TODO(Javad): handle meta space allocation failure
-                panic!("failed to mmap meta memory");
-            }
```

Comment:
This code eagerly mmaps all the metadata memory for the whole 64-bit contiguous space, which will most likely panic due to OOM.

Reply:
In the `init()` for `LockFreeImmortalSpace`, we also have code like this.

Reply:
I'm not quite sure how to deal with `LockFreeImmortalSpace`. It maps all the heap memory ahead of time. But mapping metadata memory ahead of time as well will most likely fail. Probably we shouldn't have any metadata when using  `LockFreeImmortalSpace`.

Reply:
We don't need to do anything to `LockFreeImmortalSpace`. It maps metadata only for the heap that the VM actually uses.